### PR TITLE
Fix Price display and values

### LINKED ISSUE #152: [PE-D][Tester C] Price field displays 1 decimal place instead of 2 decimal places
User guide states that it will be rounded up to 2 dp.

Upon entering a new order with price `2.999`, the application displays `3.0` instead of `3.00`.
The command entered is `order 1 d/1xRoses c/2.999 by/23-07-2024 00:00`.

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: aureliony/ped#3

### LINKED ISSUE #163: [PE-D][Tester D] App cannot launch when values below 0.005 is added and subsequently rounded to 0.
https://youtu.be/9i2l_L39aF8

This video documents the entire process of the bug.

What happens is according to your user guide, rounding will take place

However, when an input like 0.001 is made, it will be rounded to 0 and your bookkeeper only accepts positive integers. Therefore the app is unable to start after this.

Expected: Rejects values lower than 0.005 or allows cost to be saved as 0.
Actual: Fails to load because the cost is saved as 0
Steps to reproduce:
Please refer to the video.

Suggestion: Rejects values lower than 0.005 or allows cost to be saved as 0. Maybe even have a special case for rounding.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: shamesjen/ped#4

### PR DESCRIPTION
**Pull request by Jaspertzx**
The Price threshold was above 0, meaning that orders could not be 0. 
Additionally, the decimal places were not displayed to 2 decimal place.

User are confused and unable to give free orders to their loyal 
customers, and their decimal points would not display properly. 
Additionally, one user reported that the value 0 will crash the app.

Fix the Price object by formatting the toString method to format 
in 2 decimal places regardless of value, and allow 0 to be an 
accepted value.

This will fix the users that the users experienced, and prevent the 
application from crashing.

Resolves #152 
Resolves #163

### CODE REVIEW

File: src/main/java/seedu/address/model/order/Price.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public Price(String value) {
      * @return true if the price is valid
      */
     public static boolean isValidPrice(String test) {
-        return Double.parseDouble(test) > 0;
+        return Double.parseDouble(test) >= 0;
```
Comment (rertyy) [inline src/main/java/seedu/address/model/order/Price.java:37]: Did you check small negative values eg -0.005 which may have similar issue to 0.005

Comment (Jaspertzx) [reply src/main/java/seedu/address/model/order/Price.java:38]: Done, checked orders with postiive, 0, negaitve and -0.005, works great now.

**Review by kailinteoo — APPROVED** [review]
LGTM! Thanks for fixing this issue

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -168,7 +168,7 @@ public static Deadline parseDeadline(String deadline) throws ParseException {
      */
     public static Price parsePrice(String price) throws ParseException {
         requireNonNull(price);
-        String trimmedPrice = price.trim();
+        String trimmedPrice = price.replaceAll("\\s+", "");
```
Comment (rertyy) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:171]: Revert

File: src/main/java/seedu/address/model/order/Price.java [hunk 3]
```diff
@@ -13,7 +13,8 @@
 public class Price {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Price must be a positive number and can only include up to two decimal places";
+            "Price must either be 0 or a positive number. Note that any decimals after the second decimal place will "
```
Comment (rertyy) [inline src/main/java/seedu/address/model/order/Price.java:16]: Revert

File: src/main/java/seedu/address/model/order/Price.java [hunk 4]
```diff
@@ -50,7 +51,7 @@ private double truncateToTwoDecimalPlaces(double value) {
 
     @Override
     public String toString() {
-        return String.valueOf(value);
+        return String.format("%.2f", value);
```
Comment (rertyy) [inline src/main/java/seedu/address/model/order/Price.java:54]: Unsure

**Pull request merged**